components/pages/explorer/QueryActions/utils/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from 'react';
+import axios, { AxiosError, AxiosResponse, Method } from 'axios';
+import urlJoin from 'url-join';
+
+import { getConfig } from '@/config';
+import { StoredQueryObject } from '../types';
+
+const needsData = (method: Method) => ['PATCH', 'POST', 'PUT'].includes(method.toUpperCase());
+
+const useQueryStorage = ({ token = '', userId = '' }: { token?: string; userId?: string }) => {
+  const [currentEtag, setCurrentEtag] = useState('');
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [storedQueries, setStoredQueries] = useState<Record<string, StoredQueryObject>>({});
+
+  const { NEXT_PUBLIC_ARRANGER_API_URL } = getConfig();
+  const queriesStorageURL = urlJoin(NEXT_PUBLIC_ARRANGER_API_URL, 'storage', 'query');
+
+  const callQueryStorage = ({
```

Comment:
I don't love this fetcher... built it out of practicality because the auth fetcher was too demanding with the props enforcement and I could not afford getting that far out of scope with the limited deadline.

TODO!